source/as-stream.ts

```diff
@@ -1,16 +1,21 @@
 'use strict';
-const {PassThrough} = require('stream');
-const duplexer3 = require('duplexer3');
-const requestAsEventEmitter = require('./request-as-event-emitter');
-const {HTTPError, ReadError} = require('./errors');
 
-module.exports = options => {
+import {PassThrough} from 'stream';
+import duplexer3 from 'duplexer3';
+import requestAsEventEmitter from './request-as-event-emitter';
+import {HTTPError, ReadError} from './errors';
+import {Options, Response, Defaults, RequestEmitter} from './utils/types';
+
+module.exports = (options: Options & Defaults['options']) => {
 	const input = new PassThrough();
 	const output = new PassThrough();
 	const proxy = duplexer3(input, output);
 	const piped = new Set();
 	let isFinished = false;
 
+	// @todo
+	// RetryDescriptor#retries looks like it could be either a number (according to `defaults`) or a function
+	// that returns a number. Once confirmed with maintainers, this could be a bug or a bad typing on my end.
 	options.retry.retries = () => 0;
```

Comment:
This is the piece of code that fails the build. According to the [doc](https://github.com/sindresorhus/got#retry) `retries` can be `number | () => number`.

TS complains because in `defaults` we set this value to be `2` but we quickly override it here regardless of it's initial value. Once I understand more how this is used, maybe I'll be able to fix the typings.